src/lib/containers/entity/page/action_menu/EntityActionMenu.tsx

```diff
@@ -0,0 +1,234 @@
+ * Find all visible actions in the action configuration that are not in the layout.
+ */
+function getUnmappedActions(
+  actionConfiguration: ActionConfigurationMap,
+  layout: EntityActionMenuLayout,
+): ActionConfigurationMap {
+  return Object.fromEntries(
+    Object.entries(actionConfiguration).filter(([action, config]) => {
+      // We don't care about actions that are not configured or should not be visible
+      if (!config || !config.visible) {
+        return false
+      }
+
+      const allLayoutActions: ActionViewProps[] = [
+        ...layout.buttonActions,
+        ...layout.primaryMenuActions.flat(),
+        ...layout.downloadMenuActions.flat(),
+      ]
+      // Action is unmapped if it's visible and the layout doesn't have it
+      return !allLayoutActions.some(
+        layoutAction => layoutAction.action === action,
+      )
+    }),
+  )
+}
+
+/**
+ * Given a group of dropdown menu items and an action configuration,
+ * return a list of dropdown menu items with non-visible and non-configured actions removed.
+ * @param group
+ * @param actionConfiguration
+ */
+function mapAndFilterItemsInMenuGroup(
+  group: ActionViewProps[],
+  actionConfiguration: ActionConfigurationMap,
+): DropdownMenuItem[] {
+  return group.reduce(
+    (itemAcc: DropdownMenuItem[], actionViewProps: ActionViewProps) => {
+      const configForAction = actionConfiguration[actionViewProps.action]
+      // Only show the item if it's configured + visible
+      if (configForAction && configForAction.visible) {
+        itemAcc.push({
+          text: configForAction.text ?? actionViewProps.action,
+          onClick: configForAction.onClick,
+          href: configForAction.href,
+          tooltipText: configForAction.tooltipText,
+          disabled: configForAction.disabled,
+          icon: actionViewProps.icon,
+          textSx: actionViewProps.textSx,
+          iconSx: actionViewProps.iconSx,
+        })
+      }
+      return itemAcc
+    },
+    [],
+  )
+}
+
+/**
+ * Given a list of groups of dropdown menu actions, returns a list of dropdown menu item
+ * groups where non-visible actions have been removed, and groups with no visible actions
+ * have been removed.
+ * @param menuActions
+ * @param actionConfiguration
+ */
+function mapAndFilterMenuGroups(
+  menuActions: ActionViewProps[][],
+  actionConfiguration: ActionConfigurationMap,
+): DropdownMenuItem[][] {
+  return menuActions.reduce(
+    (groupAcc: DropdownMenuItem[][], group: ActionViewProps[]) => {
+      const itemGroup = mapAndFilterItemsInMenuGroup(group, actionConfiguration)
+      // Only show the group if it contains visible items
+      if (itemGroup.length > 0) {
+        groupAcc.push(itemGroup)
+      }
+      return groupAcc
+    },
+    [],
+  )
+}
+
+/**
+ * The EntityActionMenu renders a menu that displays the actions that can be invoked on an Entity page.
+ */
+export default function EntityActionMenu(props: EntityActionMenuProps) {
```

Comment:
The `ComplexMenu` component is designed to be the reusable part, since a different menu might not need the complexity of action configurations and filtering Actions based on visibility. But we could rename this if that comes up.